PIL/Image.py

```diff
@@ -2279,10 +2279,14 @@ def open(fp, mode="r"):
     filename = ""
     if isPath(fp):
         filename = fp
-    elif sys.version_info >= (3, 4):
```

Comment:
Before, the pathlib stuff was only run when `isPath(fp)` is False (and Python >= 3.4).

Now, it's always run, whether `isPath(fp)` is False or True.

Should the `try`/`except` be put in an `else` case, so it's only run when `isPath(fp)` is False?